rep-I0005.rst

```diff
@@ -0,0 +1,164 @@
+  int seq_number: 4 bytes, identifying this request
+  float x, y, z, qx, qy, qz, qw: 4 bytes each, for the requested
+  Cartesian position and quaternion rotation
+  float translational_speed: 4 bytes, the speed to be applied to the
+  translational part
+  float rotational_speed: 4 bytes, the speed to be applied to the
+  rotational part
+
+Cartesian Move Reply
+--------------------
+The robot controller should reply to each Cartesian move request with
+a reply that the request can be initiated. Immediately upon receipt of
+the move request, the robot controller should reply with this message
+with the reply_type set to 3 (EXECUTING), or 2 (FAILURE) if
+the move request is not possible. When the motion has finished, the
+robot controller should reply with the reply_type set to 1 (SUCCESS),
+or 3 (FAILURE) if the move did not finish where requested.
+
+Format of a Cartesian move reply::
+
+  int length: 4 bytes, constant value 4 x 4 = 16
+  int message_type: 4 bytes, constant value 31
+  int comm_type; 4 bytes, constant value 3, REPLY
+  int reply_type: 4 bytes, 1 = SUCCESS, 2 = FAILURE, 3 = EXECUTING
+  int seq_number: 4 bytes, identifying this request
+
+Cartesian Move State
+---------------------
+The robot controller should continually send back Cartesian move state
+messages as the move progresses, with the current pose and speeds of
+the robot. The format is the same as for the request message, except
+that the comm_type is 1, TOPIC. 
+
+Format of a Cartesian move state message::
+
+  int length: 4 bytes, constant value 13 x 4 = 52
+  int message_type: 4 bytes, constant value 31
+  int comm_type; 4 bytes, constant value 1, TOPIC
+  int reply_type: 4 bytes, ignored, any value, typically 0
+  int seq_number: 4 bytes, identifying this request
+  float x, y, z, qx, qy, qz, qw: 4 bytes each, for the requested
+  Cartesian position and quaternion rotation
+  float translational_speed: 4 bytes, the speed to be applied to the
+  translational part
+  float rotational_speed: 4 bytes, the speed to be applied to the
+  rotational part
+
+Object State
+============
+A robot may have objects under its control, such as grippers or
+fixtures or other devices tied to its controller via auxiliary
+input/output lines. For these devices, state about the object can be
+reported with the object state message.
+
+For each object, the format of its information is::
+
+  int id: 4 bytes, a unique identifier known to the application
+  float x, y, z, qx, qy, qz, qw: 4 bytes each, for the Cartesian
+  position and orientation of the object
+
+A single object state message will include header information plus one
+or more object information sections::
+ 
+  int length: 4 bytes, variable value should be size of the message
+  excluding the 4 length bytes
+  int message_type: 4 bytes, constant value 40
+  int comm_type: 4 bytes, constant value 1, TOPIC
+  int reply_type: 4 bytes, N/A
+  int seq_number: 4 bytes
+  int number: 4 bytes, the number of objects that follow
```

Comment:
In implementing the multi-robot interface, we have come across issues in the byte array library that make it difficult to have variable length messages.  @JeremyZoss, can provide some insight here.  This will be fixed.  I'm curious if you guys ran into the same issues.